engines/wintermute/ad/ad_actor_3dx.cpp

```diff
@@ -0,0 +1,2490 @@
+						//_gameRef->QuickMessageForm("%d, %d -> %f, %f, %f", x, y, pos.x, pos.y, pos.z);
+						goTo3D(pos);
+						if (strcmp(name, "GoToAsync") != 0) {
+							script->waitForExclusive(this);
+						}
+					}
+				}
+			}
+		} else {
+			if (_path2D) {
+				_path2D->reset();
+			}
+			if (_path3D) {
+				_path3D->reset();
+			}
+		}
+
+		stack->pushNULL();
+		return true;
+	}
+
+	//////////////////////////////////////////////////////////////////////////
+	// GoToObject / GoToObjectAsync
+	//////////////////////////////////////////////////////////////////////////
+	else if (strcmp(name, "GoToObject") == 0 || strcmp(name, "GoToObjectAsync") == 0) {
+		stack->correctParams(1);
+		ScValue *Val = stack->pop();
+		if (!Val->isNative()) {
+			script->runtimeError("actor.%s method accepts an entity refrence only", name);
+			stack->pushNULL();
+			return true;
+		}
+		AdObject *Obj = (AdObject *)Val->getNative();
+
+		if (!Obj || Obj->getType() != OBJECT_ENTITY) {
+			script->runtimeError("actor.%s method accepts an entity refrence only", name);
+			stack->pushNULL();
+			return true;
+		}
+
+		AdEntity *Ent = (AdEntity *)Obj;
+
+		AdGame *adGame = (AdGame *)_gameRef;
+
+		bool GoToNeeded = true;
+
+		if (Ent->getWalkToX() == 0 && Ent->getWalkToY() == 0) {
+			GoToNeeded = isGoToNeeded(Ent->_posX, Ent->_posY);
+		} else {
+			GoToNeeded = isGoToNeeded(Ent->getWalkToX(), Ent->getWalkToY());
+		}
+
+		if (!GoToNeeded) {
+			// no goto needed, but we still want to turn
+			if (Ent->getWalkToX() != 0 || Ent->getWalkToY() != 0) {
+				turnTo(dirToAngle(Ent->getWalkToDir()));
+				if (strcmp(name, "GoToObjectAsync") != 0) {
+					script->waitForExclusive(this);
+				}
+			}
+
+			if (_path2D) {
+				_path2D->reset();
+			}
+			if (_path3D) {
+				_path3D->reset();
+			}
+
+			stack->pushNULL();
+			return true;
+		}
+
+		if (adGame->_scene->_2DPathfinding) {
+			if (Ent->getWalkToX() == 0 && Ent->getWalkToY() == 0) {
+				goTo2D(Ent->_posX, Ent->_posY);
+			} else {
+				goTo2D(Ent->getWalkToX(), Ent->getWalkToY(), dirToAngle(Ent->getWalkToDir()));
+			}
+
+			if (strcmp(name, "GoToObjectAsync") != 0) {
+				script->waitForExclusive(this);
+			}
+		} else {
+			if (adGame->_scene->_sceneGeometry) {
+				Math::Vector3d pos;
+
+				if (adGame->_scene->_sceneGeometry->convert2Dto3DTolerant(Ent->getWalkToX(), Ent->getWalkToY(), &pos)) {
+					if (Ent->getWalkToX() == 0 && Ent->getWalkToY() == 0) {
+						goTo3D(pos);
+					} else {
+						goTo3D(pos, dirToAngle(Ent->getWalkToDir()));
+					}
+
+					if (strcmp(name, "GoToObjectAsync") != 0) {
+						script->waitForExclusive(this);
+					}
+				}
+			}
+		}
+		stack->pushNULL();
+		return true;
+	}
+
+	//////////////////////////////////////////////////////////////////////////
+	// TurnTo / TurnToAsync
+	//////////////////////////////////////////////////////////////////////////
+	else if (strcmp(name, "TurnTo") == 0 || strcmp(name, "TurnToAsync") == 0) {

```

Comment:
use scummvm version of strcmp.

Reply:
ScummVM port is also using strcmp. Still switch for scumm_stricmp?

Reply:
hmm, weird. ok then, don't switch